python/importer/main.py

```diff
@@ -0,0 +1,86 @@
+"""HFP Analytics data importer"""
+import azure.functions as func
+from azure.storage.blob import BlobServiceClient, BlobBlock
+from io import StringIO
+import os
+import csv
+import zstandard
+from datetime import datetime, timedelta
+import pytz
+from dateutil import parser
+import psycopg2 as psycopg
+from common.utils import get_conn_params
+from .run_analysis import main as run_analysis
+
+# TODO: import other event types as well when needed.
+event_types_to_import = ['DOC', 'DOO']
+
+def main(dataImporter: func.TimerRequest):
+    imported_successfully = False
+    conn = psycopg.connect(**get_conn_params())
+    try:
+        with conn:
+            with conn.cursor() as cur:
+                imported_successfully = import_data_to_db(cur=cur)
+                print(f' imported_successfully {imported_successfully}')
+    finally:
+        conn.close()
+        if imported_successfully == True:
+            print("Import done successfully, running analysis.")
+            run_analysis()
+
+def import_data_to_db(cur):
+    yesterday = datetime.now() - timedelta(1)
+    yesterday = datetime.strftime(yesterday, '%Y-%m-%d')
+    hfp_storage_container_name = os.getenv('HFP_STORAGE_CONTAINER_NAME')
+    hfp_storage_connection_string = os.getenv('HFP_STORAGE_CONNECTION_STRING')
+    service = BlobServiceClient.from_connection_string(conn_str=hfp_storage_connection_string)
+    result = service.find_blobs_by_tags(f"@container='{hfp_storage_container_name}' AND min_oday <= '{yesterday}' AND max_oday >= '{yesterday}'")
+
+    blob_names = []
+    for i, r in enumerate(result):
+        # File format is e.g. 2022-05-11T00-3_ARR.csv.zst
+        # extract event type as we know what the format is:
+        type = r.name.split('T')[1].split('_')[1].split('.')[0]
+        if type in event_types_to_import:
+            print(r.name)
+            blob_names.append(r.name)
+
+    imported_successfully = True
+
+    blob_index = 0
+    try:
+        for blob in blob_names:
+            print(f'Importing blob: {blob}')
+            blob_client = service.get_blob_client(container="hfp-v2-test", blob=blob)
+            storage_stream_downloader = blob_client.download_blob()
+            read_imported_data_to_db(cur=cur, downloader=storage_stream_downloader)
+            blob_index += 1
+            # Limit downloading all the blobs when developing.
+            if os.getenv('IS_DEBUG') == 'True' and blob_index > 5:
+                return imported_successfully
+    except Exception as e:
+        imported_successfully = False
+        print(f'Error in reading blob chunks: {e}')
+
+    return imported_successfully
+
+def read_imported_data_to_db(cur, downloader):
+    compressed_content = downloader.content_as_bytes()
+    reader = zstandard.ZstdDecompressor().stream_reader(compressed_content)
+    bytes = reader.readall()
+    hfp_dict_reader = csv.DictReader(StringIO(bytes.decode('utf-8')))
+    import_io = StringIO()
```

Comment:
I had to get rid of reading in chunks as I think the chunk size was different zstd frame so it could not decompress the chunk. Well, one 15 min VP file should be max 300 MB so memory overflow shouldn't be a problem.